### ISSUE TITLE
Support AWS KMS signing

### ISSUE THREAD

**Issue by ChevronTango** [theupdateframework/go-tuf#525]
Right now the only way of signing your metadata files is by using a local key file. Many users would prefer to store sensitive keys in KMS. tools like ['tuftool'](https://tuf-repo-cdn.sigstore.dev/targets.json) allow for specifying an [AWS KMS id](https://github.com/awslabs/tough/tree/665d23c40a613f50527b4913b2faa8070dfdf652/tough-kms) (eg. `tuftool root sign --key aws-kms:///<my-kms-id>`) and using that to sign the various role metadata files, allowing for a more safe environment. It would be great if the go CLI equally supported this functionality and use case.

**Comment by udf2457** [comment]
I agree more should be done to improve support for offline storage of keys. From my perspective (#427), PKCS#11 support would be far more useful, that would mean you could integrate smartcards, yubikeys etc.   I assume AWS KMS should be able to talk PKCS#11 too, or maybe they save that for AWS CloudHSM ?

**Comment by rdimitrov** [comment]
Closing as this is not in the scope for our v2 roadmap (it was relevant to the legacy version of this project though).
